backend/lib/edgehog/containers/image/changes/deploy_image_on_device.ex

```diff
@@ -0,0 +1,40 @@
+#
+# This file is part of Edgehog.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Edgehog.Containers.Image.Changes.DeployImageOnDevice do
+  @moduledoc false
+  use Ash.Resource.Change
+
+  alias Edgehog.Containers
+  alias Edgehog.Devices
+
+  @impl Ash.Resource.Change
+  def change(changeset, _opts, context) do
+    %{tenant: tenant} = context
+
+    Ash.Changeset.after_action(changeset, fn _changeset, deployment ->
```

Comment:
We could be precise when naming variables to avoid any confusion: `image_deployment` instead of `deployment`.

```suggestion
    Ash.Changeset.after_action(changeset, fn _changeset, image_deployment ->
```